File: android/app/src/main/java/com/galaxyrn/logger/GxyLogger.java
package com.galaxyrn.logger;

import android.util.Log;
import io.sentry.SentryLevel;

public class GxyLogger {

    // Log levels
    public static final int VERBOSE = 0;
    public static final int DEBUG = 1;
    public static final int INFO = 2;
    public static final int WARN = 3;
    public static final int ERROR = 4;

    // Default configuration
    private static final String DEFAULT_TAG = "GalaxyLogger";

    // VERBOSE level logging
    public static void v(String message) {
        v(DEFAULT_TAG, message);
    }

    public static void v(String tag, String message) {
        Log.v(tag, GxyLoggerUtils.formatMessage(message));
    }

    public static void v(String tag, String message, Throwable throwable) {
        Log.v(tag, GxyLoggerUtils.formatMessage(message), throwable);
    }

    // DEBUG level logging
    public static void d(String message) {
        d(DEFAULT_TAG, message);
    }

    public static void d(String tag, String message) {
        Log.d(tag, GxyLoggerUtils.formatMessage(message));
    }

    public static void d(String tag, String message, Throwable throwable) {
        Log.d(tag, GxyLoggerUtils.formatMessage(message), throwable);
    }

    // INFO level logging
    public static void i(String message) {
        i(DEFAULT_TAG, message);
    }

    public static void i(String tag, String message) {
        Log.i(tag, GxyLoggerUtils.formatMessage(message));
    }

    public static void i(String tag, String message, Throwable throwable) {
        Log.i(tag, GxyLoggerUtils.formatMessage(message), throwable);
    }

    // WARN level logging
    public static void w(String message) {
        w(DEFAULT_TAG, message);
    }

    public static void w(String tag, String message) {
        Log.w(tag, GxyLoggerUtils.formatMessage(message));
        SentryUtils.reportToSentry(SentryLevel.WARNING, tag, message, null);
    }

    public static void w(String tag, String message, Throwable throwable) {
        Log.w(tag, GxyLoggerUtils.formatMessage(message), throwable);
        SentryUtils.reportToSentry(SentryLevel.WARNING, tag, message, throwable);
    }

    // ERROR level logging
    public static void e(String message) {
        e(DEFAULT_TAG, message);
    }

    public static void e(String tag, String message) {
        Log.e(tag, GxyLoggerUtils.formatMessage(message));
        SentryUtils.reportToSentry(SentryLevel.ERROR, tag, message, null);
    }

    public static void e(String tag, String message, Throwable throwable) {
        Log.e(tag, GxyLoggerUtils.formatMessage(message), throwable);
        SentryUtils.reportToSentry(SentryLevel.ERROR, tag, message, throwable);
    }
}

File: ios/GalaxyRN/Logger/GxyLogger.swift
import Foundation
import UIKit
import os.log

/**
 * Custom Logger for GalaxyRN iOS Application
 * Provides structured logging with different levels
 */
class GxyLogger {
    
    // Log levels
    static let VERBOSE = 0
    static let DEBUG = 1
    static let INFO = 2
    static let WARN = 3
    static let ERROR = 4
    
    // Default configuration
    private static let DEFAULT_TAG = "GalaxyRN"
    
    // MARK: - VERBOSE level logging
    
    static func v(_ message: String) {
        v(DEFAULT_TAG, message)
    }
    
    static func v(_ tag: String, _ message: String) {
        os_log("%@", log: OSLog.default, type: .debug, GxyLoggerUtils.formatMessage(message))
    }
    
    static func v(_ tag: String, _ message: String, _ error: Error?) {
        let formattedMessage = GxyLoggerUtils.formatMessage(message)
        if let error = error {
            os_log("%@ - Error: %@", log: OSLog.default, type: .debug, formattedMessage, error.localizedDescription)
        } else {
            os_log("%@", log: OSLog.default, type: .debug, formattedMessage)
        }
    }
    
    // MARK: - DEBUG level logging
    
    static func d(_ message: String) {
        d(DEFAULT_TAG, message)
    }
    
    static func d(_ tag: String, _ message: String) {
        os_log("%@", log: OSLog.default, type: .debug, GxyLoggerUtils.formatMessage(message))
    }
    
    static func d(_ tag: String, _ message: String, _ error: Error?) {
        let formattedMessage = GxyLoggerUtils.formatMessage(message)
        if let error = error {
            os_log("%@ - Error: %@", log: OSLog.default, type: .debug, formattedMessage, error.localizedDescription)
        } else {
            os_log("%@", log: OSLog.default, type: .debug, formattedMessage)
        }
    }
    
    // MARK: - INFO level logging
    
    static func i(_ message: String) {
        i(DEFAULT_TAG, message)
    }
    
    static func i(_ tag: String, _ message: String) {
        os_log("%@", log: OSLog.default, type: .info, GxyLoggerUtils.formatMessage(message))
    }
    
    static func i(_ tag: String, _ message: String, _ error: Error?) {
        let formattedMessage = GxyLoggerUtils.formatMessage(message)
        if let error = error {
            os_log("%@ - Error: %@", log: OSLog.default, type: .info, formattedMessage, error.localizedDescription)
        } else {
            os_log("%@", log: OSLog.default, type: .info, formattedMessage)
        }
    }
    
    // MARK: - WARN level logging
    
    static func w(_ message: String) {
        w(DEFAULT_TAG, message)
    }
    
    static func w(_ tag: String, _ message: String) {
        os_log("%@", log: OSLog.default, type: .default, GxyLoggerUtils.formatMessage(message))
    }
    
    static func w(_ tag: String, _ message: String, _ error: Error?) {
        let formattedMessage = GxyLoggerUtils.formatMessage(message)
        if let error = error {
            os_log("%@ - Error: %@", log: OSLog.default, type: .default, formattedMessage, error.localizedDescription)
        } else {
            os_log("%@", log: OSLog.default, type: .default, formattedMessage)
        }
    }
    
    // MARK: - ERROR level logging
    
    static func e(_ message: String) {
        e(DEFAULT_TAG, message)
    }
    
    static func e(_ tag: String, _ message: String) {
        os_log("%@", log: OSLog.default, type: .error, GxyLoggerUtils.formatMessage(message))
    }
    
    static func e(_ tag: String, _ message: String, _ error: Error?) {
        let formattedMessage = GxyLoggerUtils.formatMessage(message)
        if let error = error {
            os_log("%@ - Error: %@", log: OSLog.default, type: .error, formattedMessage, error.localizedDescription)
        } else {
            os_log("%@", log: OSLog.default, type: .error, formattedMessage)
        }
    }
}
